io-sim-classes/src/Control/Monad/Class/MonadSTM.hs

```diff
@@ -44,7 +46,7 @@ class (MonadFork m, Monad (Tr m)) => MonadSTM m where
   -- The STM primitives
   type TVar m :: * -> *
 
-  atomically   :: Tr m a -> m a
+  atomically   :: HasCallStack => Tr m a -> m a
```

Comment:
Will this mean that every user of `atomically` will have to state that constraint?

Why not leave it as is and just export `wrapBlockedIndefinitely`?

Reply:
No, users of `atomically` will not be affected. They can state `HasCallStack` if they want to grow the recorded call stack but they don't have to. By having it here it means that no matter where `atomically` is called, _if_ we get that "blocked indefinitely" exception we are told where it came from without having to do anything else. Again, very useful for debugging; "blocked indefinitely" exceptions can be hard to track down; now, without doing anything else, you get a line number to go with it.